### ISSUE TITLE
Opening a PR to Add Support for Images with Non-standard Number of Channels for the ResNet models. 

### ISSUE THREAD

**Issue by RKHashmani** [pytorch/vision#6822]
### 🚀 The feature

I would like to open a pull request to add a simple feature I added.

It is simply replacing the hard-coded "3" in [torchvision/models/resnet.py](https://github.com/pytorch/vision/blob/main/torchvision/models/resnet.py)'s ResNet Class with an argument variable titled "in_channels". 

This will allow users to immediately create a ResNet models that supports a variety of initial input channels.

### Motivation, pitch

This feature allows images with non-standard channels to be easily used with PyTorch's ResNet implementations. For example, grayscale (in_channels = 1) or domain-specific feature maps can directly be treated as an input to a created ResNet model.

I myself am working with some scientific data, and our detector outputs image-like data with 2 channels. I thought this simple addition might make things easier and quicker for myself, as well as for people working on other kinds of image-like data.

### Alternatives

Currently, for my simpler ResNet models I simply replace the initial self.conv1 layer after creating an instance of a model. For example:
`self.model = torch.hub.load('pytorch/vision:v0.10.0', 'resnet18', pretrained=False, num_classes=1)`
`self.model.conv1 = nn.Conv2d(in_channels=1, out_channels=64, kernel_size=7, stride=2, padding=3, bias=False)`

For some more complicated models, I simply clone the vision repository, make the changes manually, and use this local version.

**Comment by RKHashmani** [comment]
Oh I see. That makes sense, since this seems to be a simple addition. As such, would it be possible for me to help contribute to this when the time comes for the planned revamp?

**Comment by datumbox** [comment]
Sounds good. I think the plan is to create a design document with the proposal as a whole. We want to avoid just adding things one by one without reviewing the API as a whole. It would be great to ping you once the proposal is ready to a) give your feedback and b) provide some help in terms of contributions. :)

**Comment by RKHashmani** [comment]
That sounds great, I'd be happy to help/contribute in any way. Please feel free to ping me whenever the time comes. I'll go ahead and close this thread now. Take care. :)
